akka-http-core/src/main/scala/akka/http/scaladsl/model/ws/UpgradeToWebSocket.scala

```diff
@@ -88,6 +108,18 @@ trait UpgradeToWebSocket extends jm.ws.UpgradeToWebSocket {
     subprotocol: String): HttpResponse =
     handleMessages(createScalaFlow(inSink, outSource), subprotocol = Some(subprotocol))
 
+  /**
+   * Java API
+   */
+  def handleMessagesWithMat[Mat](handlerFlow: Graph[FlowShape[jm.ws.Message, jm.ws.Message], Mat]): (HttpResponse, Future[Mat]) =
```

Comment:
ah ok - so should I change it in the java dsl as well? Basically everywhere should return the ```Pair``` case class